CHANGELOG.md

```diff
@@ -13,6 +13,7 @@ v4.0 (in development)
 
 Converting from v3.x to v4.0
 -----------------------------
+- A significant difference between v3.3 and 4.0 is the naming of dependencies. Unique names were not enforced in 3.3, which lead to undefined behavior. In 4.0, Component pathnames must be unique. That is a Component must be unique with respect to its peers. A Model named *model* cannot have multiple subcomponents with the name *toes* either as bodies or joints, because the pathname */model/toes* will not uniquely identify the Component. However, multiple *toes* bodies can be used as long as they are not subcomponents of the same Component. For example, a *device* Component with a *toes* Body will have no issues since this *toes* Body has a unique pathname, */model/device/toes*, which in unambiguous. One could also create multi-legged model, where each leg is identical, with *hip*. *knee* and *ankle* joints and *upper*, *lower* and *foot* bodies, but each being a unique because of Leg Component that contains the leg subcomponents, is uniquely named like */model/leg1* and */model/leg4/* and thus all of their subcomponents are unique, e.g.: */model/leg1/knee* vs. */model/leg4/knee*.        
```

Comment:
Thanks for adding this helpful documentation. Some typos:

- `*hip*. *knee* and *ankle*` -> `*hip*, *knee* and *ankle*` (period -> comma)
- `being a unique` -> `being unique`


I think the second part could be more brief ("However, a model with two `knee` components is fine if the absolute pathnames are unique, e.g., `/model/left_leg/knee` and `/model/right_leg/knee`.").